src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNITcpHandle.cs

```diff
@@ -182,31 +183,59 @@ public SNITCPHandle(string serverName, int port, long timerExpire, object callba
             _status = TdsEnums.SNI_SUCCESS;
         }
 
-        private static async Task<Socket> ConnectAsync(string serverName, int port)
+        private static Socket Connect(string serverName, int port, TimeSpan timeout)
         {
-            IPAddress[] addresses = await Dns.GetHostAddressesAsync(serverName).ConfigureAwait(false);
-            IPAddress targetAddrV4 = Array.Find(addresses, addr => (addr.AddressFamily == AddressFamily.InterNetwork));
-            IPAddress targetAddrV6 = Array.Find(addresses, addr => (addr.AddressFamily == AddressFamily.InterNetworkV6));
-            if (targetAddrV4 != null && targetAddrV6 != null)
+            IPAddress[] ipAddresses = Dns.GetHostAddresses(serverName);
+            IPAddress serverIPv4 = null;
+            IPAddress serverIPv6 = null;
+            foreach (IPAddress ipAdress in ipAddresses)
```

Comment:
This loop iterates over the entire array even after we've found both IPs, and may have multiple assignments to the same variables. Array.Find is also more compact.

Reply:
> This loop iterates over the entire array even after we've found both IPs

Maximum length of the returned array would be single digit integer. It returned IP address array which size is 2 (IPv4, IPv6) in my environment. I don't see performance decrease for that.

>, and may have multiple assignments to the same variables.

Technically a host have have multiple IPv4 or IPv6 address depending on the number of NIC. However it is not recommended, and also rare case in modern networking. Even though the returned IP array has multiple IPv4 or IPv6, the length of the array would be single digit constant, which does not affect performance. 

> Array.Find is also more compact.

I don't see a reason to loop the array twice by `Array.Find()` when we can achieve the same thing by looping once.

Therefore, Won't fix.